AMW_angular/io/src/app/resources/resource-type-add.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, EventEmitter, inject, Input, Output, Signal } from '@angular/core';
+import { ResourceType } from '../resource/resource-type';
+import { ResourceTypesService } from './resource-types.service';
+import {
+  NgbActiveModal,
+  NgbDropdown,
+  NgbDropdownItem,
+  NgbDropdownMenu,
+  NgbDropdownToggle,
+} from '@ng-bootstrap/ng-bootstrap';
+import { ModalHeaderComponent } from '../shared/modal-header/modal-header.component';
+import { ButtonComponent } from '../shared/button/button.component';
+import { FormsModule } from '@angular/forms';
+import { ResourceTypeRequest } from '../resource/resource-type-request';
+
+@Component({
+  selector: 'app-resource-type-add',
+  standalone: true,
+  imports: [
+    ModalHeaderComponent,
+    ButtonComponent,
+    FormsModule,
+    NgbDropdown,
+    NgbDropdownItem,
+    NgbDropdownMenu,
+    NgbDropdownToggle,
+  ],
+  templateUrl: './resource-type-add.component.html',
+})
+export class ResourceTypeAddComponent {
+  @Input() resourceType: ResourceType;
+  @Output() saveResourceType: EventEmitter<ResourceTypeRequest> = new EventEmitter<ResourceTypeRequest>();
+  private resourceTypesService = inject(ResourceTypesService);
+  parentResourceTypeName: string;
+  rootResourceTypes: Signal<ResourceType[]> = this.resourceTypesService.rootResourceTypes;
+  parentId: number;
+
+  constructor(public activeModal: NgbActiveModal) {}
+
+  cancel() {
+    this.activeModal.close();
+    this.resourceTypesService.refreshData();
+  }
+
+  save() {
```

Comment:
revalidate input and only emit something if it is valid. Otherwise, there is always a request send on any click